### PR TITLE
[Sources] Refactored component to use contents.id

### PR DESCRIPTION
**Pull request by Hindelburg**
…s.id instead of contents.get("id").

Fixes Issue: #5738 

### Summary of Changes

* Modified SourceTree.js to use content.id instead of content.get("id").

### Test Plan

We ran the testing suite using Yarn and we opened the debugger and tested the primary features.

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

### CODE REVIEW

**Comment by darkwing** [comment]
One side effect I'm seeing is that the Blackbox icon is no longer displayed in the tree when a source has been blackboxed.

File: src/components/PrimaryPanes/SourcesTree.js [hunk 1]
```diff
@@ -233,7 +224,7 @@ class SourcesTree extends Component<Props, State> {
     const menuOptions = [];
 
     if (!isDirectory(item)) {
-      const source = item.contents.get("url");
+      const source = item.contents.url;
```
Comment (lukaszsobek) [inline src/components/PrimaryPanes/SourcesTree.js]: Maybe name this `sourceURL` and change the corresponding `copyToTheClipboard` as this is what we are actually getting here.

**Comment by lukaszsobek** [comment]
When I inspect the pr I have 10 tests failing, namely:

```
 FAIL  src/components/tests/SourcesTree.spec.js

  ● SourcesTree › on receiving new props › up
dates list items › updates list items if show
nSource changes

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [{"sourceId": "server1.conn13.child1/41"}]
    But it was called with:
      [{"sourceId": undefined}]

      at Object.it (src/components/tests/SourcesTree.spec.js:160:38)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)


  ● SourcesTree › focusItem › update the focused item

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [{"sourceId": "server1.conn13.child1/39"}]
    But it was called with:
      [{"sourceId": undefined}]

      at Object.it (src/components/tests/SourcesTree.spec.js:193:36)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)


  ● SourcesTree › renderItem › should show source item with source icon

    TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Receivedtype undefined

      at Url.parse (url.js:106:11)
      at urlParse (url.js:100:13)
      at getExtension (src/utils/sources-tree/utils.js:103:231)
      at getSourceClassnames (src/utils/source.js:401:931)
      at SourcesTree.getIcon (src/components/PrimaryPanes/SourcesTree.js:414:1205)
      at SourcesTree.renderItem (src/components/PrimaryPanes/SourcesTree.js:414:3079)
      at renderItem (src/components/tests/SourcesTree.spec.js:605:19)
      at Object.it (src/components/tests/SourcesTree.spec.js:359:20)


  ● SourcesTree › renderItem › should show source item with source icon with focus

    TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Receivedtype undefined

      at Url.parse (url.js:106:11)
      at urlParse (url.js:100:13)
      at getExtension (src/utils/sources-tree/utils.js:103:231)
      at getSourceClassnames (src/utils/source.js:401:931)
      at SourcesTree.getIcon (src/components/PrimaryPanes/SourcesTree.js:414:1205)
      at SourcesTree.renderItem (src/components/PrimaryPanes/SourcesTree.js:414:3079)
      at renderItem (src/components/tests/SourcesTree.spec.js:605:19)
      at Object.it (src/components/tests/SourcesTree.spec.js:365:20)


  ● SourcesTree › renderItem › should not show domain item when the projectRoot exists

    TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Receivedtype undefined

      at Url.parse (url.js:106:11)
      at urlParse (url.js:100:13)
      at getExtension (src/utils/sources-tree/utils.js:103:231)
      at getSourceClassnames (src/utils/source.js:401:931)
      at SourcesTree.getIcon (src/components/PrimaryPanes/SourcesTree.js:414:1205)
      at SourcesTree.renderItem (src/components/PrimaryPanes/SourcesTree.js:414:3079)
      at renderItem (src/components/tests/SourcesTree.spec.js:605:19)
      at Object.it (src/components/tests/SourcesTree.spec.js:394:20)


  ● SourcesTree › renderItem › should show menu on contextmenu of an item

    TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Receivedtype undefined

      at Url.parse (url.js:106:11)
      at urlParse (url.js:100:13)
      at getExtension (src/utils/sources-tree/utils.js:103:231)
      at getSourceClassnames (src/utils/source.js:401:931)
      at SourcesTree.getIcon (src/components/PrimaryPanes/SourcesTree.js:414:1205)
      at SourcesTree.renderItem (src/components/PrimaryPanes/SourcesTree.js:414:3079)
      at renderItem (src/components/tests/SourcesTree.spec.js:605:19)
      at Object.it (src/components/tests/SourcesTree.spec.js:403:41)


  ● SourcesTree › renderItem › should focus on and select item on click

    TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Receivedtype undefined

      at Url.parse (url.js:106:11)
      at urlParse (url.js:100:13)
      at getExtension (src/utils/sources-tree/utils.js:103:231)
      at getSourceClassnames (src/utils/source.js:401:931)
      at SourcesTree.getIcon (src/components/PrimaryPanes/SourcesTree.js:414:1205)
      at SourcesTree.renderItem (src/components/PrimaryPanes/SourcesTree.js:414:3079)
      at renderItem (src/components/tests/SourcesTree.spec.js:605:19)
      at Object.it (src/components/tests/SourcesTree.spec.js:415:7)


  ● SourcesTree › selectItem › should selectitem with no children

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [{"sourceId": "server1.conn13.child1/39"}]
    But it was called with:
      [{"sourceId": undefined}]

      at Object.it (src/components/tests/SourcesTree.spec.js:450:36)
          at new Promise (<anonymous>)
      at Promise.resolve.then.el (node_modules/p-map/index.js:46:16)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)


  ● SourcesTree › selectItem › should selectitem on enter onKeyDown event

    expect(jest.fn()).toHaveBeenCalledWith(expected)

    Expected mock function to have been called with:
      [{"sourceId": "server1.conn13.child1/39"}]
    But it was called with:
      [{"sourceId": undefined}]

      at Object.it (src/components/tests/SourcesTree.spec.js:468:36)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)


  ● SourcesTree › getPath › should return path for blackboxedboxed item

    expect(received).toEqual(expected)

    Expected value to equal:
      "http://mdn.com/blackboxed.js/blackboxed.js/update"
    Received:
      "http://mdn.com/blackboxed.js/blackboxed.js/"

      at Object.it (src/components/tests/SourcesTree.spec.js:564:20)
          at new Promise (<anonymous>)
      at Promise.resolve.then.el (node_modules/p-map/index.js:46:16)
          at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)


Test Suites: 1 failed, 94 passed, 95 total
Tests:       10 failed, 658 passed, 668 total
Snapshots:   272 passed, 272 total
```

**Comment by lukaszsobek** [comment]
@Hindelburg Let me know what I can do to support you in getting this done :) There are a lot of bits flying around here.

**Pull request merged**